commons/zenoh-codec/src/core/zint.rs

```diff
@@ -112,14 +112,24 @@ where
     fn write(self, writer: &mut W, mut x: u64) -> Self::Output {
         writer.with_slot(VLE_LEN, move |buffer| {
             let mut len = 0;
-            let mut b = x as u8;
-            while x > 0x7f {
-                buffer[len] = b | 0x80;
+            while (x & !0x7f) != 0 {
+                // SAFETY: buffer is guaranteed to be VLE_LEN long where VLE_LEN is
+                //         the maximum number of bytes a VLE can take once encoded.
+                //         I.e.: x is shifted 7 bits to the right every iteration,
+                //         the loop is at most VLE_LEN iterations.
+                unsafe {
+                    *buffer.get_unchecked_mut(len) = (x as u8) | 0x80;
```

Comment:
same here, I would suggest specifying explicitly the type of 0x80 literal as 0x80u8

Reply:
Done